mantidproject/mantid
# EA plotting widget

**Description of work.**
This pull request adds the plotting widget to the new elemental analysis widget

**To test:**
1. Before starting Mantid go to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L24 , add Muon/Elemental_Analysis_2.py to the end of the line and build Mantid.

2. Download the test files from PR #29958 and add to User Directories.

3. Open the elemental analysis 2 GUI and load 9997.

4. Workspace to plot can be chosen using checkboxes in the Analyse column in the grouping tab.
![plotting widget_editted](https://user-images.githubusercontent.com/59264035/124392763-4356f980-dcef-11eb-94de-4d727cad562d.png)

5.  Combobox labelled 1 is used to change the the spectrum plotted
6.  Combobox labelled 2 is used to change how tiled plots should be seperated
7.  Plot raw checkbox  is used to plot rebinned workspaces and should spawn a warning window if unchecked and no rebinnned workspaces in plotted groups


Fixes #29384. 


*This does not require release notes* because  new GUI is not being released yet



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
